pkg/i2gw/provider.go

```diff
@@ -51,13 +70,14 @@ type ProviderConf struct {
 type Provider interface {
 	CustomResourceReader
 	ResourceConverter
+	ResourceFilter
 }
 
 type CustomResourceReader interface {
 
 	// ReadResourcesFromCluster reads custom resources associated with
 	// the underlying Provider implementation from the kubernetes cluster.
-	ReadResourcesFromCluster(ctx context.Context, customResources interface{}) error
+	ReadResourcesFromCluster(ctx context.Context, cl client.Client, customResources interface{}) error
```

Comment:
Because when reading from a file we have no necessity to have a client. Before this change, in such a context and without a cluster up and running the client instantiation was failing, leading to an error in the tool.

Reply:
Left a comment above related to it. Can we retain the client within ProviderConf and fix the logic so reading from file logic would not use it?